seed/seed16.go

```diff
@@ -124,11 +124,13 @@ func (s *seed16) SetParallelism(int) {
 	// ignored
 }
 
-func (s *seed16) addSnap(sn *internal.Snap16, pinnedTrack string, cache map[string]*Snap, tm timings.Measurer) (*Snap, error) {
+func (s *seed16) addSnap(sn *internal.Snap16, essType snap.Type, pinnedTrack string, handler SnapHandler, cache map[string]*Snap, tm timings.Measurer) (*Snap, error) {
 	path := filepath.Join(s.seedDir, "snaps", sn.File)
 
+	_, defaultHandler := handler.(defaultSnapHandler)
+
 	seedSnap := cache[path]
-	if seedSnap == nil {
+	if seedSnap == nil || !defaultHandler {
```

Comment:
The `!defaultHandler` condition is not clear to me, can you add a comment in the code?

Reply:
is to ignore the cache if a handler is passed in, as explained in the doc comment of the methods. If we use the cache we wouldn't call the handler for the cached snap which could be unexpected. I'll add comments in code